mirrord/operator/src/setup.rs

```diff
@@ -691,6 +701,41 @@ impl OperatorRoleBinding {
     }
 }
 
+#[derive(Debug)]
+pub struct OperatorSecretRole(Role);
+
+impl OperatorSecretRole {
+    pub fn new(namespace: &OperatorNamespace, kafka_splitting: bool) -> Option<Self> {
```

Comment:
```suggestion
    pub fn new(namespace: &OperatorNamespace) -> Self {
```
And call it like `kafka_splitting.then(|| todo!("constructor here"))`. Wdyt?

Reply:
oh nice, yeah I wasn't a fan of a `new()` function returning `Option` - this is way nicer